src/select.rs

```diff
@@ -20,6 +22,9 @@ pub struct Select<T> {
     items: Vec<RadioButton<T>>,
     cursor: usize,
     initial_value: Option<T>,
+    enable_filter_mode: bool,
+    input_filter: StringCursor,
+    filtered_items: Vec<RadioButton<T>>,
```

Comment:
@antoinePutmans  BTW, it's up to you, but if you have motivation, I was thinking, that it's worth combining 3 fields into a separate structure like `struct FilterMode`, then `filter: Option<FilterMode>`, and then you can just pass some events into it like `self.filter.on(&key)`, `self.filter.render()`, instead of big `if-else` code blocks now, it would make the change patch cleaner and it would **dissect** filter logic from the main selection logic. The only catch is that it's essentially an improvement, a subjective thing, and not a requirement to make this PR pass (which I believe is almost done, 99%).

Reply:
Yes it could be a great improvement for the readability, maybe I can implement that on a next PR ? :) 
I still have to implement the max-select-list length so I will do both at the same time if its ok for you